Tasks/UseNodeV1/proxyutil.ts

```diff
@@ -0,0 +1,44 @@
+import * as taskLib from 'azure-pipelines-task-lib/task';
+import * as qs from 'querystring';
+import * as url from 'url';
+
+interface ICurlProxy {
+    variable: string,
+    setting: string
+}
+
+function toCurlProxy(proxyCfg: taskLib.ProxyConfiguration): ICurlProxy {
+    let curlProxy: ICurlProxy;
+    if (proxyCfg) {
+        if (proxyCfg && proxyCfg.proxyUrl) {
+            taskLib.debug(`using proxy ${proxyCfg.proxyUrl}`);
+            const parsedUrl = url.parse(proxyCfg.proxyUrl);
+            const httpEnvVarName: string = parsedUrl.protocol === 'https:'? "https_proxy" : "http_proxy";
```

Comment:
should protocal comparison be case insensitive? e.g. parsedUrl.protocol.toUpper()

Reply:
No, url.parse ensures it will be lowercase.